sim/side.ts

```diff
@@ -243,14 +243,14 @@ export class Side {
 	allies(all?: boolean) {
 		// called during the first switch-in, so `active` can still contain nulls at this point
 		let allies = this.activeTeam().filter(ally => ally);
-		if (!all) allies = allies.filter(ally => !ally.fainted);
+		if (!all) allies = allies.filter(ally => !ally.fainted && !!ally.hp);
```

Comment:
This seems bad...

Reply:
Oh, wait, I misread this. I think this is probably fine? We probably need to figure out what `pokemon.fainted` actually means, though, and what it means when it differs from `!pokemon.hp`.

Also this is probably redundant? `ally => !!ally.hp` should mean the same thing, right?

Reply:
Yeah, I'm saying, that would be

```suggestion
		if (!all) allies = allies.filter(ally => !!ally.hp);
```

Reply:
b95adb3ae4607e6 didn't add a redundant check either, for the record.